.github/scripts/validateWorkflowSchemas.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+##########################################################
+#    Validate GitHub action and workflow yaml schemas    #
+##########################################################
+
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)"
+readonly SCRIPT_DIR
+GITHUB_DIR="$(dirname "$SCRIPT_DIR")"
+readonly GITHUB_DIR
+
+source "$SCRIPT_DIR/shellUtils.sh"
+
+title 'Validating the Github Actions and workflows using the json schemas provided by (https://www.schemastore.org/json/)'
+
+# Create a temporary directory for schemas
+TEMP_SCHEMA_DIR="$(mktemp -d)"
+trap 'rm -rf "$TEMP_SCHEMA_DIR"' EXIT
+
+function download_schema() {
+    [[ $1 = 'github-action.json' ]] && SCHEMA_NAME='GitHub Action' || SCHEMA_NAME='GitHub Workflow'
+    info "Downloading $SCHEMA_NAME schema..."
+    if curl "https://json.schemastore.org/$1" --output "$TEMP_SCHEMA_DIR/$1" --silent; then
+        success "Successfully downloaded $SCHEMA_NAME schema!"
+        echo
+    else
+        error "Failed downloading $SCHEMA_NAME schema"
+        exit 1
+    fi
+}
+
+# Download the up-to-date json schemas for github actions and workflows
+download_schema 'github-action.json' || exit 1
+download_schema 'github-workflow.json' || exit 1
```

Comment:
You don't need `|| exit 1` since your function is already existing, but you can simplify even further by using a `for` loop

Reply:
done